lib/quantum/job_broadcaster.ex

```diff
@@ -52,35 +47,32 @@ defmodule Quantum.JobBroadcaster do
     {:noreply, to_send, %{state | buffer: remaining}}
   end
 
-  def handle_cast({:add, %Job{state: :active} = job}, state) do
+  def handle_cast({:add, %Job{state: :active, name: job_name} = job}, %{jobs: jobs} = state) do
     Logger.debug(fn ->
-      "[#{inspect(Node.self())}][#{__MODULE__}] Adding job #{inspect(job.name)}"
+      "[#{inspect(Node.self())}][#{__MODULE__}] Adding job #{inspect(job_name)}"
     end)
 
-    {:noreply, [{:add, job}], put_in(state[:jobs][job.name], job)}
+    {:noreply, [{:add, job}], %{state | jobs: Map.put(jobs, job_name, job)}}
   end
 
-  def handle_cast({:add, %Job{state: :inactive} = job}, state) do
+  def handle_cast({:add, %Job{state: :inactive, name: job_name} = job}, %{jobs: jobs} = state) do
     Logger.debug(fn ->
-      "[#{inspect(Node.self())}][#{__MODULE__}] Adding job #{inspect(job.name)}"
+      "[#{inspect(Node.self())}][#{__MODULE__}] Adding job #{inspect(job_name)}"
     end)
 
-    {:noreply, [], put_in(state[:jobs][job.name], job)}
+    {:noreply, [], %{state | jobs: Map.put(jobs, job_name, job)}}
   end
 
   def handle_cast({:delete, name}, %{jobs: jobs} = state) do
     Logger.debug(fn ->
       "[#{inspect(Node.self())}][#{__MODULE__}] Deleting job #{inspect(name)}"
     end)
 
-    cond do
```

Comment:
The job is not deleted from state anymore when it is inactive.

Reply:
And before my the change it wasn't removed either.

Maybe tests should be updated to check expected behavior.

Reply:
Oh yeah, you're right about that. I'm going to add a test.